### PR TITLE
#160445634 Refactor and clean up actions

### PR DESCRIPTION
**Pull request by cosmas28**
#### What does this PR do?
This PR cleans up the action and action tests to make the code DRY.
#### Description of Task to be completed?
There is a lot of repeated blocks of code in the actions and action tests, therefore, I created reusable functions to avoid repetition.
#### Where should the reviewer start?
[Place link to file here]()
#### How should this be manually tested?
N/A
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#160445634](https://www.pivotaltracker.com/story/show/160445634)
#### Screenshots (if appropriate)
N/A

### CODE REVIEW

File: src/actions/errorAction.test.js [hunk 1]
```diff
@@ -1,11 +1,17 @@
-import configureMockStore from 'redux-mock-store';
-import thunk from 'redux-thunk';
-import * as actions from './incidentAction';
-import * as types from './actionTypes';
+// third-party libraries
```
Comment (andela-oakinseye) [inline src/actions/errorAction.test.js]: What's happening on this line @cosmas28

File: src/actions/errorAction.test.js [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
 import moxios from 'moxios';
 
-const middlewares = [thunk];
-const mockStore = configureMockStore(middlewares);
+// helpers
```
Comment (andela-oakinseye) [inline src/actions/errorAction.test.js]: There is no need for this comment too.

File: src/actions/errorAction.test.js (hunk 1 above)
Comment (cosmas28) [reply src/actions/errorAction.test.js]: It's a comment, do you mean I should remove it?

**Review by theartisancodes — APPROVED** [review]

File: src/actions/staffAction.jsx [hunk 3]
```diff
@@ -23,15 +20,13 @@ export const fetchStaff = () => {
...
-  return { type: SEARCH_USER, staff, isError: false };
-};
+export const searchUserSuccess = staff => ({
+  type: SEARCH_USER, staff, isError: false
+});
 
 export const searchUsers = query => {
   return dispatch => {
-    return axios
-      .get(`${config.API_URL}/users/search/?q=${query}`)
```
Comment (andela-oakinseye) [inline src/actions/staffAction.jsx]: Why was `config.API_URL` removed?

Comment (cosmas28) [reply src/actions/staffAction.jsx]: I created a reusable helper function for Axios which prevented the repetition of `config.API_URL`, token` and `header` variables.

**Pull request merged**